source/common/ssl/context_config_impl.cc

```diff
@@ -101,6 +109,13 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::auth::CommonTlsContex
   }
 }
 
+ContextConfigImpl::~ContextConfigImpl() {
+  if (tls_certficate_provider_.get() != nullptr && secret_callback_ != nullptr) {
+    tls_certficate_provider_->removeUpdateCallback(*secret_callback_);
+    secret_callback_ = nullptr;
```

Comment:
nit: del, this should not matter within the destructor.

Reply:
Thanks! Removed.